src/WebJobs.Extensions.CosmosDB/CosmosDBSqlResolutionPolicy.cs

```diff
@@ -36,18 +35,44 @@ public string TemplateBind(PropertyInfo propInfo, Attribute resolvedAttribute, B
             SqlParameterCollection paramCollection = new SqlParameterCollection();
 
             // also build up a dictionary replacing '{token}' with '@token' 
-            IDictionary<string, string> replacements = new Dictionary<string, string>();
+            IDictionary<string, object> replacements = new Dictionary<string, object>();
             foreach (var token in bindingTemplate.ParameterNames.Distinct())
             {
-                string sqlToken = $"@{token}";
-                paramCollection.Add(new SqlParameter(sqlToken, bindingData[token]));
-                replacements.Add(token, sqlToken);
+                object tokenObject = bindingData[token];
+                if (tokenObject is string tokenString)
+                {
+                    string sqlToken = GetSqlParameterName(token);
```

Comment:
This case seems like a subset of the other in terms of logic, just thinking out loud, but what if we had a single logic that created the replacements and the difference is that when the param is a `string` we are sending an input Dictionary of 1 item? Probably @brettsam has a more valuable feedback 😄